### PR TITLE
[BREAKING] Remove HTMLAudioElement fallback from sound system

### PR DESCRIPTION
**Pull request by willeastcott**
### Summary

The engine used to ship a parallel `HTMLAudioElement`-based implementation of its sound system for browsers without Web Audio API support, gated by `hasAudioContext()`. Every browser the engine targets (Chrome 10+, Firefox 25+, Safari 6+, Edge 12+) has had Web Audio for years, so the fallback was dead code that only ever covered IE — which the engine no longer supports.

This PR removes the fallback and all the branching / duplicated prototype definitions that existed to support it.

### Changes

- Deleted `src/platform/sound/capabilities.js` (existed solely to gate the fallback).
- `SoundInstance`: flattened the constructor to the Web Audio branch and removed the ~230-line `Object.assign(SoundInstance.prototype, …)` override block (duplicate `play`/`pause`/`resume`/`stop`/`setExternalNodes`/`clearExternalNodes`/`getExternalNodes`/`_createSource`/`_onLoadedMetadata`/`_onTimeUpdate`/`_onManagerDestroy`) and the redefined `volume` / `pitch` / `sound` / `currentTime` accessors.
- `SoundInstance3d`: removed the `fallOff` JS distance-function fallback and the redefined `position` / `maxDistance` / `refDistance` / `rollOffFactor` / `distanceModel` accessors.
- `AudioHandler`: removed the IE detection IIFE and the `new Audio()` + `canplaythrough` branch in `_createSound`; now uses Web Audio only.
- `Sound`: removed the public `audio` property; constructor now accepts only an `AudioBuffer`.
- `SoundComponentSystem.context`: no longer emits the "Audio context is not supported" warning — it just returns `this.manager.context`, which already returns `null` if no `AudioContext`/`webkitAudioContext` is available.

`webkitAudioContext` handling in `SoundManager` is intentionally left alone.

Net: **+80 / -523 lines** across 6 files.

### Public API changes (breaking)

`Sound.audio` is removed.

**Before**
```js
// A Sound could wrap an HTMLAudioElement if Web Audio wasn't available
const sound = new pc.Sound(audioElement);
sound.audio; // HTMLAudioElement
```

**After**
```js
// Sound wraps an AudioBuffer only
const sound = new pc.Sound(audioBuffer);
sound.buffer; // AudioBuffer
```

`SoundComponentSystem#context` no longer logs a warning or returns `null` on the theoretical "no Web Audio" browser. It now returns the underlying `SoundManager.context`, which was already `null` in the edge case where neither `AudioContext` nor `webkitAudioContext` exist. Behaviour is effectively unchanged on every supported browser.

### Test plan

- [x] `npm run lint` — clean
- [x] `npm run build:types` — succeeds; verified `Sound.audio`, `HTMLAudioElement`, and `hasAudioContext` no longer appear in `build/playcanvas.d.ts`
- [x] `npm test` — 1640 passing, 0 failing
- [x] Manual smoke: run the `sound-*` examples and confirm 2D / 3D playback still works

### CODE REVIEW

File: src/platform/sound/instance3d.js [hunk 1]
```diff
@@ -1,8 +1,6 @@
 import { Debug } from '../../core/debug.js';
-import { math } from '../../core/math/math.js';
 import { Vec3 } from '../../core/math/vec3.js';
-import { DISTANCE_EXPONENTIAL, DISTANCE_INVERSE, DISTANCE_LINEAR } from './constants.js';
-import { hasAudioContext } from './capabilities.js';
+import { DISTANCE_LINEAR } from './constants.js';
 import { SoundInstance } from './instance.js';
```
[the remark this answers is not in the document]
Comment (willeastcott) [reply src/platform/sound/instance3d.js:4]: JSDoc `{@link Name}` resolves against the engine's public-API symbol table (the top-level exports in `src/index.js`), not against the file's local imports. `DISTANCE_INVERSE` and `DISTANCE_EXPONENTIAL` are both exported from `src/platform/sound/constants.js` and re-exported from the main index, so the links still resolve in typedoc.

The sibling file `src/framework/components/sound/component.js` uses the same pattern today (imports only `DISTANCE_LINEAR` but references `{@link DISTANCE_INVERSE}` / `{@link DISTANCE_EXPONENTIAL}` in JSDoc), and the `Docs` CI check on this PR — which runs typedoc — passes. No change needed here.

**Review by mvaligursky — APPROVED** [review]
Nice!
